internal/services/backoff.go

```diff
@@ -47,6 +58,9 @@ func (b *Backoff) Delay() (time.Duration, error) {
 		return 0, &MaxAttemptsExceeded{}
 	}
 	delay := time.Duration(b.scale * 1 << b.attempt)
+	if delay > b.max {
```

Comment:
what about overflow from the line above?

Reply:
if delay < 0 panic(...) would be enough for me. The ensuing kube restart would put the code back onto the right track.

Reply:
@k-wall good point, I changed the code a little bit for checking overflow.